vpr/src/route/route_net.tpp

```diff
@@ -0,0 +1,520 @@
+            conn_delay_budget.min_delay = budgeting_inf.get_min_delay_budget(net_id, target_pin);
+            conn_delay_budget.short_path_criticality = budgeting_inf.get_crit_short_path(net_id, target_pin);
+            conn_delay_budget.routing_budgets_algorithm = router_opts.routing_budgets_algorithm;
+        }
+
+        profiling::conn_start();
+
+        // build a branch in the route tree to the target
+        auto sink_flags = timing_driven_route_sink(router,
+                                                   net_list,
+                                                   net_id,
+                                                   itarget,
+                                                   target_pin,
+                                                   cost_params,
+                                                   router_opts,
+                                                   tree,
+                                                   spatial_route_tree_lookup,
+                                                   router_stats,
+                                                   budgeting_inf,
+                                                   routing_predictor,
+                                                   choking_spots,
+                                                   is_flat);
+
+        flags.retry_with_full_bb |= sink_flags.retry_with_full_bb;
+
+        if (!sink_flags.success) {
+            flags.success = false;
+            return flags;
+        }
+
+        profiling::conn_finish(size_t(route_ctx.net_rr_terminals[net_id][0]),
+                               size_t(sink_rr),
+                               pin_criticality[target_pin]);
+
+        ++router_stats.connections_routed;
+    } // finished all sinks
+
+    ++router_stats.nets_routed;
+    profiling::net_finish();
+
+    /* For later timing analysis. */
+
+    // may have to update timing delay of the previously legally reached sinks since downstream capacitance could be changed
+    update_net_delays_from_route_tree(net_delay,
+                                      net_list,
+                                      net_id,
+                                      timing_info,
+                                      pin_timing_invalidator);
+
+    if (router_opts.update_lower_bound_delays) {
+        for (int ipin : remaining_targets) {
+            connections_inf.update_lower_bound_connection_delay(net_id, ipin, net_delay[ipin]);
+        }
+    }
+
+    VTR_ASSERT_MSG(g_vpr_ctx.routing().rr_node_route_inf[tree.root().inode].occ() <= rr_graph.node_capacity(tree.root().inode), "SOURCE should never be congested");
+
+    VTR_LOGV_DEBUG(f_router_debug, "Routed Net %zu (%zu sinks)\n", size_t(net_id), num_sinks);
+    router.empty_rcv_route_tree_set(); // ?
+
+    flags.success = true;
+    return flags;
+}
+
+template<typename ConnectionRouter>
+inline NetResultFlags timing_driven_pre_route_to_clock_root(ConnectionRouter& router,
```

Comment:
Comment this if you can.

Reply:
It routes to a special sink that is not in the netlist, and which represents the start point of a global clock network.